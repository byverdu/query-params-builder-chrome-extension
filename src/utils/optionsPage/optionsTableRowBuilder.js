/**
 *
 * @param {string} id
 * @param {string} bundleName
 * @param {string} urlParamKey
 * @returns string
 */
export function optionsTableRowBuilder(id, bundleName, urlParamKey) {
  return `
  <tr id="${id}">
  <td class="contentEditable" data-initial-value="${bundleName}" data-value-type="bundleName" contentEditable="true">
    <span>${bundleName}</span>
  </td>
  <td class="contentEditable" data-initial-value="${urlParamKey}" data-value-type="urlParamKey" contentEditable="true">
    <span>${urlParamKey}</span>
  </td>
  <td>
    <button data-bundle-id="${id}" class="btn btn-outline-danger delete-bundle">
      Delete
    </button>
  </td>
</tr>
  `;
}
