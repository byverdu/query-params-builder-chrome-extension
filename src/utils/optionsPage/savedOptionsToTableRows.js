import { optionsTableRowBuilder } from './optionsTableRowBuilder';

/**
 * @param {import('../../types/index.js').ExtensionOptions[]} optionsToBuild
 */
export function savedOptionsToTableRows(optionsToBuild) {
  const tbody = document.querySelector('.selected_bundles tbody');

  if (!tbody) {
    throw new Error(
      'No .selected_bundles tbody selector present on the document'
    );
  }

  for (const option of optionsToBuild) {
    const tr = optionsTableRowBuilder(
      option.id,
      option.bundleName,
      option.urlParamKey
    );

    tbody.innerHTML += tr;
  }
}
