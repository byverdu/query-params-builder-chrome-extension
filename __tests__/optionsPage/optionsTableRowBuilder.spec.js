import { optionsTableRowBuilder } from '../../src/utils/optionsPage/optionsTableRowBuilder';

describe('optionsTableRowBuilder', () => {
  it('should be defined', () => {
    expect(optionsTableRowBuilder).toBeInstanceOf(Function);
  });

  it('should render the <tr>', () => {
    const markup = optionsTableRowBuilder('12345', 'api key', 'apiKey');

    expect(markup).toMatchSnapshot();
  });
});
