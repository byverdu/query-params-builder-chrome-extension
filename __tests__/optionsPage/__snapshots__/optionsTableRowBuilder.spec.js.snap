// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optionsTableRowBuilder should render the <tr> 1`] = `
"
  <tr id="12345">
  <td class="contentEditable" data-initial-value="api key" data-value-type="bundleName" contentEditable="true">
    <span>api key</span>
  </td>
  <td class="contentEditable" data-initial-value="apiKey" data-value-type="urlParamKey" contentEditable="true">
    <span>apiKey</span>
  </td>
  <td>
    <button data-bundle-id="12345" class="btn btn-outline-danger delete-bundle">
      Delete
    </button>
  </td>
</tr>
  "
`;
